ts/components/SessionPasswordPrompt.tsx

```diff
@@ -50,164 +44,174 @@ function pushToastError(id: string, description: string) {
+  return (
+    <div className={clsx(showResetElements && 'session-modal__button-group')}>
+      {showResetElements && (
+        <>
+          <SessionButton
+            text={window.i18n('clearDevice')}
+            buttonColor={SessionButtonColor.Danger}
+            buttonType={SessionButtonType.Simple}
+            onClick={onShowClearDataView}
+          />
+        </>
+      )}
+      {!loading && (
+        <SessionButton
+          text={showResetElements ? window.i18n('tryAgain') : window.i18n('done')}
+          buttonType={SessionButtonType.Simple}
+          onClick={initLogin}
+          disabled={loading}
+        />
+      )}
+    </div>
+  );
+}
 
-    autoBind(this);
+async function onLogin(passPhrase: string, increaseErrorCount: () => void, onFinished: () => void) {
+  // Note: we don't trim the password anymore. If the user entered a space at the end, so be it.
+  try {
+    await window.onLogin(passPhrase);
+  } catch (error) {
+    // Increment the error counter and show the button if necessary
+    increaseErrorCount();
+
+    if (error && isString(error)) {
+      pushToastError('onLogin', error);
+    } else if (error?.message && isString(error.message)) {
+      pushToastError('onLogin', error.message);
+    }
   }
+  onFinished();
+}
 
-  public componentDidMount() {
-    setTimeout(() => {
-      this.inputRef?.focus();
-    }, 100);
-  }
+function ClearDataDescription() {
+  return (
+    <p>
+      <Localizer token="clearDeviceDescription" />
+    </p>
+  );
+}
 
-  public render() {
-    const isLoading = this.state.loading;
-    const spinner = isLoading ? <SessionSpinner loading={true} /> : null;
-    const featureElement = this.state.clearDataView ? (
-      <p>
-        <Localizer token="clearDeviceDescription" />
-      </p>
-    ) : (
-      <div className="session-modal__input-group">
-        <input
-          type="password"
-          id="password-prompt-input"
-          defaultValue=""
-          placeholder={window.i18n('passwordEnter')}
-          onKeyUp={this.onKeyUp}
-          ref={input => {
-            this.inputRef = input;
-          }}
-        />
-      </div>
-    );
-
-    return (
-      <SessionWrapperModal
-        title={this.state.clearDataView ? window.i18n('clearDevice') : window.i18n('passwordEnter')}
-      >
-        {spinner || featureElement}
-        <TextPleaseWait isLoading={isLoading} />
-        {this.state.clearDataView
-          ? this.renderClearDataViewButtons()
-          : this.renderPasswordViewButtons()}
-      </SessionWrapperModal>
-    );
-  }
+const PasswordPrompt = ({
+  onEnterPressed,
+  onPasswordChange,
+}: {
+  onPasswordChange: (password: string) => void;
+  onEnterPressed: () => void;
+}) => {
+  const inputRef = useRef<HTMLInputElement>(null);
 
-  public onKeyUp(event: any) {
-    switch (event.key) {
-      case 'Enter':
-        this.initLogin();
-        break;
-      default:
-    }
-    event.preventDefault();
-  }
+  useTimeoutFn(() => {
```

Comment:
Is this still necessary? Alternatively you can use the SessionInput here which has an autofocus prop?